plugins/browser-plugin-media-tracking/src/helperFunctions.ts

```diff
@@ -114,12 +49,41 @@ export function dataUrlHandler(url: string): string {
   return url;
 }
 
-export function getDuration(el: HTMLAudioElement | HTMLVideoElement): number | null {
-  const duration = el.duration;
-  // A NaN value is returned if duration is not available, or Infinity if the media resource is streaming.
-  if (isNaN(duration) || duration === Infinity) {
-    return null;
+type TimeRange = { start: number; end: number };
+export function timeRangesToObjectArray(t: TimeRanges): TimeRange[] {
+  const out: TimeRange[] = [];
+  for (let i = 0; i < t.length; i++) {
+    out.push({ start: t.start(i), end: t.end(i) });
   }
+  return out;
+}
 
-  return duration;
+type TextTrack = {
+  label: string;
+  language: string;
+  kind: string;
+  mode: string;
+};
+export function textTrackListToJson(textTrackList: TextTrackList): TextTrack[] {
+  return Object.keys(textTrackList).map((_, i) => {
+    return {
+      label: textTrackList[i].label,
+      language: textTrackList[i].language,
+      kind: textTrackList[i].kind,
+      mode: textTrackList[i].mode,
+    };
+  });
+}
+
+export function setConfigDefaults(config: Config): Config {
+  const defaults = {
+    captureEvents: HTMLEventTypes,
```

Comment:
Do we need to set the defaults? Does it change anything compared to not specifying `captureEvents` if not passed by user? If we don't specify `captureEvents`, the tracker will track all events tracked by the media-tracking plugin.

If there is a reason to specify the default `captureEvents`, please add ping event type to them. That is tracked by default in the media plugin but because we override this here, it is not tracked anymore.

Reply:
Yep we can get rid of that now. I've added ping and percentprogress to `HTML5MediaEventType` also.